minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/UpdateConfigurationService.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.minifi.bootstrap.service;
+
+import static java.util.Optional.ofNullable;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.util.Optional;
+import org.apache.commons.io.IOUtils;
+import org.apache.nifi.minifi.bootstrap.MiNiFiCommandState;
+import org.apache.nifi.minifi.bootstrap.RunMiNiFi;
+import org.apache.nifi.minifi.bootstrap.configuration.ConfigurationChangeListener;
+import org.apache.nifi.minifi.bootstrap.configuration.differentiators.Differentiator;
+import org.apache.nifi.minifi.bootstrap.configuration.differentiators.WholeConfigDifferentiator;
+import org.apache.nifi.minifi.bootstrap.util.ByteBufferInputStream;
+import org.apache.nifi.minifi.bootstrap.util.ConfigTransformer;
+import org.apache.nifi.util.NiFiProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UpdateConfigurationService {
+
+    private static final Logger logger = LoggerFactory.getLogger(UpdateConfigurationService.class);
+    private static final String UPDATED_CONFIG_FILE_NAME = "config-updated.yml";
+
+    private final Differentiator<ByteBuffer> differentiator;
+    private final RunMiNiFi runMiNiFi;
+    private final ConfigurationChangeListener miNiFiConfigurationChangeListener;
+    private final BootstrapFileProvider bootstrapFileProvider;
+
+    public UpdateConfigurationService(RunMiNiFi runMiNiFi, ConfigurationChangeListener miNiFiConfigurationChangeListener, BootstrapFileProvider bootstrapFileProvider) {
+        this.differentiator = WholeConfigDifferentiator.getByteBufferDifferentiator();
+        this.differentiator.initialize(runMiNiFi);
+        this.runMiNiFi = runMiNiFi;
+        this.miNiFiConfigurationChangeListener = miNiFiConfigurationChangeListener;
+        this.bootstrapFileProvider = bootstrapFileProvider;
+    }
+
+    public Optional<MiNiFiCommandState> handleUpdate() {
+        logger.debug("Handling configuration update");
+        Optional<MiNiFiCommandState> commandState = Optional.empty();
```

Comment:
commandState doesn't need to be optional you can set it to null here and at return you could do Optional.ofNullable.